ui/src/segments/components/form/validation/schema.js

```diff
@@ -2,6 +2,62 @@
 import * as yup from "yup";
 
 const segmentNameRegex = /^[A-Za-z\d][\w\d \-()#$%&:.]*[\w\d\-()#$%&:.]$/;
+export const segmentConfigSchema = yup.lazy((obj) =>
+  yup.object().shape(
+    Object.keys(obj).reduce((acc, key) => {
+      return {
+        ...acc,
+        [key]: yup.mixed()
+          .when("$segmenterTypes", (segmenterTypes) => {
+            switch ((segmenterTypes[key] || "").toUpperCase()) {
```

Comment:
Using `|| ""` for when the `type` value is undefined, as with segmenters that are deactivated from the project settings later on.